src/TenancyServiceProvider.php

```
@@ -102,6 +113,34 @@ public function register(): void
+     * Ensure globalCache uses the central connection for database cache stores.
+     *
+     * A freshly built CacheManager creates database stores using the current default connection, which
+     * DatabaseTenancyBootstrapper switches to the tenant connection. Since global cache should always be
+     * central, reset those stores back to their configured connection, falling back to the central one.
+     */
+    protected function makeDatabaseCacheStoresCentral(CacheManager $manager): void
+    {
+        $centralConnection = $this->app['config']['tenancy.database.central_connection'];
+
+        foreach ($this->app['config']['cache.stores'] ?? [] as $name => $store) {
+            $notAValidDatabaseStore = ! is_array($store) || ($store['driver'] ?? null) !== 'database';
```

PHPMD (2.15.0)

[warning] Avoid excessively long variable names like $notAValidDatabaseStore. Keep variable name length under 20. (undefined)

(LongVariable)